dvc/objects/db/base.py

```diff
@@ -75,6 +76,38 @@ def get(self, hash_info: "HashInfo"):
             hash_info,
         )
 
+    def _add_file(self, from_fs, from_info, to_info, move):
+        self.makedirs(to_info.parent)
+        use_move = isinstance(from_fs, type(self.fs)) and move
+        try:
+            if use_move:
+                self.fs.move(from_info, to_info)
+            elif isinstance(from_fs, LocalFileSystem):
+                if not isinstance(from_info, from_fs.PATH_CLS):
+                    from_info = from_fs.PATH_CLS(from_info)
+                self.fs.upload(from_info, to_info)
```

Comment:
Random thought: I guess we could also put a check for `self.fs` being a local one and then using `from_fs.download`?